string from
string to
---
string res